### PR TITLE
Rover: set trim to min ignoring reversal for brushed with relay

### LINKED ISSUE #13269: SERVOx_REVERSED does not work as expected for ESC type "BrushedWithRelay"
# Bug report

**Issue details**

According to documentation https://ardupilot.org/rover/docs/rover-motor-and-servo-configuration.html#testing-motor-direction
Function SERVOx_REVERSED should change the rotation direction of the motor.

When I set the SERVOx_REVERSED in the running flight controller with mission planner at first it works correctly. The relay output is inverted and the motor turns in the opposite direction.
But when I reboot the flight controller the Trim value is automatically set to 2000 causing a PWM signal output of 2000 with no throttle which makes the motor run full speed.
When I set the Trim value back to 1000 in Mission Planner manually and save it works correctly again.
But after a reboot the Trim value is again automatically set back to 2000.
The same happens after disarming and arming again. After arming again the PWM output is 2000 when no throttle is given. Changing the trim value to 1000 mannually restores the correct behaviour but only until the next disarm/rearm procedure.

**Version**
V4.0.0

**Platform**
[  ] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[x] Rover
[  ] Submarine

**Hardware type**
Kakute F7 AIO

### PR DESCRIPTION
**Pull request by IamPete1**
Rework of https://github.com/ArduPilot/ardupilot/pull/13386 from @tig987 

Fixes #13269

### CODE REVIEW

File: libraries/SRV_Channel/SRV_Channel_aux.cpp [hunk 1]
```diff
@@ -584,12 +584,12 @@ void SRV_Channels::set_trim_to_pwm_for(SRV_Channel::Aux_servo_function_t functio
...
 
-// set the trim for a function channel to min output
-void SRV_Channels::set_trim_to_min_for(SRV_Channel::Aux_servo_function_t function)
+// set the trim for a function channel to min output of the channel honnoring reverse unless ignore_reversed is true
+void SRV_Channels::set_trim_to_min_for(SRV_Channel::Aux_servo_function_t function, bool ignore_reversed)
 {
     for (uint8_t i=0; i<NUM_SERVO_CHANNELS; i++) {
         if (channels[i].function == function) {
-            channels[i].servo_trim.set(channels[i].get_reversed()?channels[i].servo_max:channels[i].servo_min);
+            channels[i].servo_trim.set((channels[i].get_reversed() & !ignore_reversed)?channels[i].servo_max:channels[i].servo_min);
```
Comment (khancyr) [inline libraries/SRV_Channel/SRV_Channel_aux.cpp:592]: &&?

Comment (IamPete1) [reply libraries/SRV_Channel/SRV_Channel_aux.cpp]: Fixed, although I don't think it makes any difference since there both bools.

**Review by khancyr — APPROVED** [review]
LGTM

**Comment by IamPete1** [comment]
https://github.com/ArduPilot/ardupilot/issues/9553

**Comment by rmackay9** [comment]
@IamPete1, can I ask how this has been tested?

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**